src/TableFooter.js

```diff
@@ -1,17 +1,44 @@
 import React from 'react'
 import PropTypes from 'prop-types'
+import paginate from 'jw-paginate'
 
 const TableFooter = ({dataArrayLength, currentPageDataLength, currentPage, entriesPerPage, onChange, dataLength}) => {
-  const pageNumbers = []
-  for (let i = 1; i <= Math.ceil(dataArrayLength / entriesPerPage); i++) {
-    pageNumbers.push(
-      i === currentPage ?
-        <li className={`current`} key={`bottom${i}`}>{currentPage}</li> :
-        <li key={`bottom${i}`}><a onClick={() => onChange(i)}>{i}</a></li>
-    )
+
+  const pageNumbersLength = Math.ceil(dataArrayLength / entriesPerPage)
+  const paginationOutput = paginate(dataArrayLength, currentPage, entriesPerPage, 3).pages
+
+  const restylePagination = (pages, totalPages) => {
```

Comment:
This function is a bit dangerous. As a habit never modify the arguments. Create a deep copy of `pages`, work with the copy and return it. Notice that below, where you call this function, both the return value and `paginationOutput` will have the same contents, and you lose the original value. I know that you know what you’re doing, but this is a practice that should apply universally. Side effects can be really insidious.